### PR TITLE
Make resizing on border click configurable

### PR DESCRIPTION
**Pull request by Deiz**
This is quite minor, but most of my border clicks are accidental (when I want to resize, I tend to use the keyboard or the modifier key) and I figured the behaviour may as well be configurable.

### CODE REVIEW

**Comment by acrisci** [comment]
What's the use case for this?

**Comment by acrisci** [comment]
Couldn't you just set the border to none?

**Comment by Deiz** [comment]
In some cases yes, but not always. I like to keep a 1px border on my terminals, and mostly run into unwanted resizes when quickly selecting a block of text at one of the edges.

Whether anybody else is interested in disabling the behaviour, I'm unsure.

**Comment by acrisci** [comment]
I'm going to oppose this change because it's a mouse-centric config directive that has a pretty good workaround.

**Comment by stapelberg** [comment]
Yeah, a new config directive is a bit much for this. If we could make this behavior achievable using mouse bindings, I’d possibly be more open to merging it.

**Comment by Deiz** [comment]
Ah, I'd completely forgotten about mouse bindings. How about adding a `--border` flag to mouse bindings? That would allow e.g. `bindsym --border button3 nop` to do precisely what I want.

**Comment by Deiz** [comment]
I noticed that some floating window actions will accept any mouse button, such as clicks on decorations (anything except for right triggers drag) and borders (all buttons trigger resize).

Presumably those should be restricted slightly to mirror their tiling counterparts (left on decorations triggers move, left/right on borders starts resize). Any opposition to that?

**Comment by acrisci** [comment]
@Deiz I had some trouble understanding that. Can you rephrase that question?

**Comment by Deiz** [comment]
Tiling windows only trigger resize on decoration and border left/right click, whereas floating windows allow any button to trigger resize. I'm proposing that floating windows ignore non-left/right clicks, like tiling windows already do.

**Comment by acrisci** [comment]
Yeah, I agree that should be made consistent.

**Comment by stapelberg** [comment]
No, we want to preserve the current behavior. I expect floating windows to be resizable and draggable with the left mouse button, and plenty of notebook users with unfortunately bad touchpads will be thankful.

**Comment by Deiz** [comment]
I'm only proposing to stop middle mouse, etc. as well as back/forward buttons from triggering floating resize.

Floating move is already restricted to the left mouse button, and I wouldn't be doing anything to it. Floating resize would continue to be triggered by left or right click on borders, or right click on decorations.

**Comment by stapelberg** [comment]
Ah. Filtering out any mouse buttons but left/right sounds reasonable.

**Comment by Deiz** [comment]
Alright, since Github doesn't support changing the target branch of a pull request, I've clobbered this PR's branch with an implementation of `--border`. It's functional, tests pass, etc., but it still needs to be documented in the user guide.

If adding `--border` is desirable, I'll update the userguide and resubmit this as a new pull request.

**Comment by stapelberg** [comment]
@acrisci what do you think? Personally I’m fine with having --border in the mouse bindings.

**Comment by acrisci** [comment]
Yeah, that seems to be a much cleaner approach to me. I'm ok with that.

File: parser-specs/config.spec [hunk 1]
```diff
@@ -339,6 +343,8 @@ state MODE_IGNORE_LINE:
 state MODE_BINDING:
   release = '--release'
       ->
+  border = '--border'
```
Comment (Airblader) [inline parser-specs/config.spec]: This also needs to go to the state MODE_BINDCOMMAND

Comment (Deiz) [inline parser-specs/config.spec]: Yeah, I corrected that locally after noticing the changes you made in #1631.

**Comment by Deiz** [comment]
Closing this as it's superseded by #1632.
